static/js/rich_search/stat_vars.tsx

```diff
@@ -31,21 +36,31 @@ function getURL(places: string[], statsVar: string): string {
   return `/tools/timeline#${params}`;
 }
 
-function StatVars({ places, statVars }: StatVarsPropType): JSX.Element {
-  if (!statVars.length) {
+export function StatVars({
+  placeName,
+  statVarInfo,
+  statVarOrder,
+}: StatVarsPropType): JSX.Element {
+  if (!statVarOrder.length) {
     return <section className="block col-12">No results found.</section>;
   }
+  const places = Object.keys(placeName);
   return (
     <section className="block col-12">
+      <div id="chart-region">
+        <ChartRegion
+          placeName={placeName}
+          statVarInfo={statVarInfo}
+          statVarOrder={statVarOrder}
+        ></ChartRegion>
+      </div>
       <ul>
-        {statVars.slice(0, 20).map((sv) => (
-          <li key={sv.dcid}>
-            <a href={getURL(places, sv.dcid)}>{sv.name}</a>
+        {statVarOrder.map((sv) => (
+          <li key={sv}>
+            <a href={getURL(places, sv)}>{statVarInfo[sv].title}</a>
           </li>
         ))}
       </ul>
     </section>
   );
 }
-
-export const MemoStatVars = memo(StatVars);
```

Comment:
i'm curious why you've chosen lodash's memo over react's? were there performance differences observed?

i also see that there is now no memoization for stat vars, but only for place names. could you comment here, or in the pr description, on some of these internal changes?

Reply:
My understanding is that React's memo is for components and controls when the rendering will happen. I added it originally since the state of the search box was always in sync with the hash url, which re-forced renders of this component unnecessarily and produced a non responsive experience. After rethinking the design, I changed the logic so that the state of the search box was only updated when the search button or the enter key were pressed, which removed the need for the memo call.

Independently, I added _.memoize (which is similar to React.useMemo but React independent) as a way to cache the network results for a backed query, since both components (StatVars and PlaceSearch) needed the output of this promise. Refactoring the logic would have made it much more complicated, and PlaceSearch will probably be removed in a future version due to the two search boxes being unified.

Let me know if that makes sense.

Reply:
Actually since the search API was updated to extract locations in a previous PR I went ahead and removed the place search bar which simplified the logic overall

Reply:
thanks for the explanation - that makes sense